### DISCUSSION TITLE
How to prefill Array Fields

### DISCUSSION THREAD

**Discussion by Eric-Juquel** [react-hook-form/react-hook-form#1712] [General] [answered]
Hi i'm new on React and react-hook-form and i wonder how can i prefill Array Fields like the one on codesnadbx :  https://codesandbox.io/s/6j1760jkjk  , for exemple to edit the form thats got current Friends array ?

**Comment by bluebill1049** [comment, marked as answer]
Take a look this codesandbox: https://codesandbox.io/s/react-hook-form-usefieldarray-vy8fv

you can use `defaultValuues` from `useForm`.

**Reply by Eric-Juquel** [reply]
thanks for your reactive answer  bluebill1049 , this way is interesting but need many changes in my own script so for the moment i finally passed conditionally my array to indexes initialState and array length to counter when current !== null and it looks like it works . but i keep this way in memory , could be very usefull . I like react-hook form a lot , you and your team do a great job !

**Reply by bluebill1049** [reply]
Thanks, @Eric-Juquel we try our best.
